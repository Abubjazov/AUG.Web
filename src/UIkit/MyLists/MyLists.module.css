.root {
  /* padding: 0 36px; */
  margin-top: 40px;
  text-align: left;
  transition: all var(--transition-delay) linear;
  animation: visible 0.5s ease-in-out;
}

.menu-closed {
  display: none;
}

.title {
  font-size: 16px;
  font-weight: 600;
  line-height: 19px;
}

.list {
  display: flex;
  flex-direction: column;
}

.list-item {
  margin-top: 20px;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
}

.list-item a {
  color: var(--blue);
}

@keyframes visible {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}
