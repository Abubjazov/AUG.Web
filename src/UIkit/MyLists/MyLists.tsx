import { FC } from 'react'

import { combineClasses as cc } from 'utils/combineClasses'

import styles from './MyLists.module.css'

interface MyListsProps {
  menuOpened: boolean
}

const MyLists: FC<MyListsProps> = ({ menuOpened }) => {
  return (
    <div className={cc([styles.root, menuOpened ? '' : styles['menu-closed']])}>
      <span className={styles['title']}>My Lists</span>

      <div className={styles['list']}>
        <span className={styles['list-item']}>
          TOP-10 Twitter Dapplets (<a href="#">Me</a>)
        </span>

        <span className={styles['list-item']}>
          Best Financial dapplets by Jack (<a href="#">Jack</a>)
        </span>

        <span className={styles['list-item']}>
          TOP-10 Essentials Dapplets (<a href="#">Me</a>)
        </span>
      </div>
    </div>
  )
}

export default MyLists
