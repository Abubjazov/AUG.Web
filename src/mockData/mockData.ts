import { IDapplet, ITag } from 'store/slices/dappletsSlice'
import { IList, IUserDapplet } from 'store/slices/userDataSlice'

export const mockUserTags: ITag[] = [
  { tagId: 'darP5Jyz8yirTMsfY9RP', tagName: 'Twitter' },
  { tagId: 'JW3UFtZ5HgATcwldsJ1T', tagName: 'Facebook' },
  { tagId: '6UKxNrMzte6RCWIlssvM', tagName: 'Top 10' },
  { tagId: 'ZbEuko9nOlQYKG6I1Efx', tagName: 'Test' },
  { tagId: '521keF1Adymn8Oo896vv', tagName: 'ToDo' },
]

export const mockUserLists: IList[] = [
  { listId: 'XfLdXIykGHsgIjdJkTaE', listName: 'My special list' },
  { listId: 'c6rClgOltChco2XnyVDp', listName: 'Financial list' },
]

export const mockUserDapplets: IUserDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    userTags: ['JW3UFtZ5HgATcwldsJ1T', '521keF1Adymn8Oo896vv'],
    dappletState: true,
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    userTags: [],
    dappletState: true,
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    userTags: 'N/A',
    dappletState: true,
  },
]

export const mockCommunityTags: ITag[] = [
  { tagId: 'XoEVTHXhfMAjIWsi4wxY', tagName: 'Top 100' },
  { tagId: '3LYUFQTbdNULwBgCkfXx', tagName: 'Finances' },
  { tagId: 'XZ2zPZqa4fPxM9mLA9xR', tagName: 'Media' },
  { tagId: 'C69B5fQ7ccW6Q1SC0yow', tagName: 'Favourites' },
]

export const mockDapplets: IDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    appOwner: 'Satoshi Nakamoto',
    circulatingSupply: '19534212',
    communityTags: ['3LYUFQTbdNULwBgCkfXx', 'C69B5fQ7ccW6Q1SC0yow'],
    date: '1230757200',
    fullDesc:
      'Bitcoin is a peer-to-peer online currency, meaning that all transactions happen directly between equal, independent network participants, without the need for any intermediary to permit or facilitate them. Bitcoin was created, according to Nakamoto’s own words, to allow “online payments to be sent directly from one party to another without going through a financial institution.”',
    fullyDilutedMarketCap: '738866583732',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-1.01',
    maxSupply: '21000000',
    name: 'Bitcoin Contracts',
    shortDesc:
      'Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. ',
    shortName: 'BTC',
    totalSupply: '19534106',
    volume: '10.9',
    volumePerMarketCap: '1.58',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHGCOPY',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    appOwner: 'XRP Ledger',
    circulatingSupply: '53615837759',
    communityTags: 'N/A',
    date: '1620939600',
    fullDesc:
      'Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.',
    fullyDilutedMarketCap: '65183420834',
    logo: '/public/images/notAvailable.svg',
    marketCap: '5.66',
    maxSupply: '100000000000',
    name: 'XRP Ledger',
    shortDesc:
      'Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.',
    shortName: 'XRP',
    totalSupply: '99988316618',
    volume: '132.26',
    volumePerMarketCap: '5.37',
  },
  {
    dappletId: 'QbWG3sKvfgFcP5RtskMp',
    appOwner: 'Realcoin',
    circulatingSupply: '85367759449',
    communityTags: [],
    date: '1388523600',
    fullDesc:
      'Launched in 2014, Tether is a blockchain-enabled platform designed to facilitate the use of fiat currencies in a digital manner. Tether works to disrupt the conventional financial system via a more modern approach to money. Tether has made headway by giving customers the ability to transact with traditional currencies across the blockchain, without the inherent volatility and complexity typically associated with a digital currency. As the first blockchain-enabled platform to facilitate the digital use of traditional currencies (a familiar, stable accounting unit), Tether has democratised cross-border transactions across the blockchain.',
    fullyDilutedMarketCap: '88708606718',
    logo: '/dappletsLogos/tether.png',
    marketCap: '0.01',
    maxSupply: 'N/A',
    name: 'Tether USDt',
    shortDesc:
      'Tether works to disrupt the conventional financial system via a more modern approach to money.',
    shortName: 'USDT',
    totalSupply: '88623656724',
    volume: '41.44',
    volumePerMarketCap: '32.16',
  },
]
