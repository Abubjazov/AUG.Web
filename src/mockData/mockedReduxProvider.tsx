import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import {
  mockCommunityTags,
  mockDapplets,
  mockUserDapplets,
  mockUserTags,
  mockUserLists,
} from './mockData'
import authReducer from '../store/slices/authSlice'
import dappletsReducer from '../store/slices/dappletsSlice'
import layoutReducer from '../store/slices/layoutSlice'
import userDataSliceReducer from '../store/slices/userDataSlice'

export const defaultMockState = {
  auth: {
    isUserAuthenticated: false,
    isInProgress: false,
    uid: undefined,
    email: null,
  },

  layout: {
    menuOpened: true,
    menuButtonsState: 0,
    dappletSettingsOpened: false,
    modalState: false,
    modalInner: undefined,
    messages: [],
  },

  userData: {
    userDapplets: mockUserDapplets,
    userTags: mockUserTags,
    userLists: mockUserLists,
    isAddingUserTag: false,
    isAddingUserList: false,
    isLoadingUserData: false,
    tagOperationGoing: [],
    dappletOperationGoing: [],
    listOperationGoing: [],
  },

  dapplets: {
    isLoadingDapplets: false,
    isNoMoreDapplets: false,
    dapplets: mockDapplets,
    tags: mockCommunityTags,
    loadFilter: {
      withLimit: 12,
      withStartAfter: undefined,
      withWhere: undefined,
    },
    lastVisible: undefined,
    orderBy: undefined,
    searchString: '',
  },
}

export const mokedStore = configureStore({
  reducer: {
    auth: authReducer,
    layout: layoutReducer,
    dapplets: dappletsReducer,
    userData: userDataSliceReducer,
  },
  preloadedState: {
    ...defaultMockState,
  },
})

export const mockedReduxProvider = ({
  children,
}: PropsWithChildren<object>) => {
  return <Provider store={mokedStore}>{children}</Provider>
}
