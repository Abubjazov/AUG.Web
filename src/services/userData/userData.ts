import { createAsyncThunk } from '@reduxjs/toolkit'
import { getAuth } from 'firebase/auth'
import {
  fireStoreGetDoc,
  fireStoreSetDoc,
} from 'services/fireStoreAPI/fireStoreAPI'
import { userDataConverter } from 'services/fireStoreDataConverters/fireStoreDataConverters'
import { RootState } from 'store/index'
import { ITag } from 'store/slices/dappletsSlice'
import {
  IUserDapplet,
  setUserDapplets,
  setUserTags,
} from 'store/slices/userDataSlice'
import { getErrorMessage } from 'utils/getErrorMessage/getErrorMessage'

export const getUserData = createAsyncThunk<
  void,
  never,
  { rejectValue: string }
>('auth/getUserData', async (_, { rejectWithValue, dispatch }) => {
  const auth = getAuth()
  const uid = auth.currentUser?.uid

  try {
    if (!uid) {
      throw new Error('An error occurred while trying to load the user data')
    }

    const userData = await fireStoreGetDoc('UsersData', uid, userDataConverter)

    dispatch(setUserDapplets(userData.userDapplets))
    dispatch(setUserTags(userData.userTags))
  } catch (e) {
    return rejectWithValue(getErrorMessage(e))
  }
})

export const addUserTag = createAsyncThunk<
  {
    userTags: ITag[]
  },
  ITag,
  { state: RootState; rejectValue: string }
>('auth/addUserTag', async (tag, { rejectWithValue, getState }) => {
  const uid = getState().auth.uid
  const state = getState().userData.userTags

  try {
    if (!uid) {
      throw new Error('An error occurred while trying to add new user tag')
    }

    const newData = {
      userTags: [...state, tag],
    }

    await fireStoreSetDoc(newData, 'UsersData', uid, { merge: true })

    return newData
  } catch (e) {
    return rejectWithValue(getErrorMessage(e))
  }
})

export const removeUserTag = createAsyncThunk<
  {
    userDapplets: IUserDapplet[]
    userTags: ITag[]
  },
  string,
  { state: RootState; rejectValue: string }
>('auth/removeUserTag', async (tagId, { rejectWithValue, getState }) => {
  const uid = getState().auth.uid
  const stateClone = structuredClone(getState().userData)

  try {
    if (!uid) {
      throw new Error('An error occurred while trying to remove user tag')
    }

    stateClone.userDapplets.map(dapplet => {
      dapplet.userTags = dapplet.userTags.filter(tag => tag !== tagId)
    })

    const newData = {
      userDapplets: stateClone.userDapplets.filter(
        dapplet => dapplet.dappletState || dapplet.userTags.length,
      ),
      userTags: stateClone.userTags.filter(myTag => myTag.tagId !== tagId),
    }

    await fireStoreSetDoc(newData, 'UsersData', uid, { merge: true })

    return newData
  } catch (e) {
    return rejectWithValue(getErrorMessage(e))
  }
})

export const installDapplet = createAsyncThunk<
  {
    userDapplets: IUserDapplet[]
  },
  string,
  { state: RootState; rejectValue: string }
>('auth/installDapplet', async (dappletId, { rejectWithValue, getState }) => {
  const uid = getState().auth.uid
  const stateClone = structuredClone(getState().userData.userDapplets)

  try {
    if (!uid) {
      throw new Error('An error occurred while trying to remove user tag')
    }

    const incomingDappletIndex = stateClone.findIndex(
      dapplet => dapplet.dappletId === dappletId,
    )

    if (incomingDappletIndex < 0) {
      stateClone.push({
        dappletId: dappletId,
        userTags: [],
        dappletState: true,
      })
    } else {
      stateClone[incomingDappletIndex].dappletState = true
    }

    const newData = {
      userDapplets: stateClone,
    }

    await fireStoreSetDoc(newData, 'UsersData', uid, { merge: true })

    return newData
  } catch (e) {
    return rejectWithValue(getErrorMessage(e))
  }
})

export const unInstallDapplet = createAsyncThunk<
  {
    userDapplets: IUserDapplet[]
  } | void,
  string,
  { state: RootState; rejectValue: string }
>('auth/unInstallDapplet', async (dappletId, { rejectWithValue, getState }) => {
  const uid = getState().auth.uid
  let stateClone = structuredClone(getState().userData.userDapplets)

  try {
    if (!uid) {
      throw new Error('An error occurred while trying to remove user tag')
    }

    const incomingDappletIndex = stateClone.findIndex(
      dapplet => dapplet.dappletId === dappletId,
    )

    if (incomingDappletIndex > -1) {
      const dappletUserTagsLength =
        stateClone[incomingDappletIndex].userTags.length

      if (dappletUserTagsLength) {
        stateClone[incomingDappletIndex].dappletState = false
      } else {
        stateClone = stateClone.filter(
          dapplet => dapplet.dappletId !== dappletId,
        )
      }

      const newData = {
        userDapplets: stateClone,
      }

      await fireStoreSetDoc(newData, 'UsersData', uid, { merge: true })

      return newData
    }
  } catch (e) {
    return rejectWithValue(getErrorMessage(e))
  }
})

export const addUserTagToDapplet = createAsyncThunk<
  {
    userDapplets: IUserDapplet[]
  },
  { dappletId: string; userTag: ITag },
  { state: RootState; rejectValue: string }
>('auth/addUserTagToDapplet', async (data, { rejectWithValue, getState }) => {
  const uid = getState().auth.uid
  const stateClone = structuredClone(getState().userData.userDapplets)

  try {
    if (!uid) {
      throw new Error('An error occurred while trying to remove user tag')
    }

    const incomingDappletIndex = stateClone.findIndex(
      dapplet => dapplet.dappletId === data.dappletId,
    )

    if (incomingDappletIndex > -1) {
      const dappletIncomingUserTagIndex = stateClone[
        incomingDappletIndex
      ].userTags.findIndex(tagId => tagId === data.userTag.tagId)

      if (dappletIncomingUserTagIndex < 0) {
        stateClone[incomingDappletIndex].userTags.push(data.userTag.tagId)
      }
    } else {
      stateClone.push({
        dappletId: data.dappletId,
        userTags: [data.userTag.tagId],
        dappletState: false,
      })
    }

    const newData = {
      userDapplets: stateClone,
    }

    await fireStoreSetDoc(newData, 'UsersData', uid, { merge: true })

    return newData
  } catch (e) {
    return rejectWithValue(getErrorMessage(e))
  }
})

export const removeUserTagFromDapplet = createAsyncThunk<
  {
    userDapplets: IUserDapplet[]
  } | void,
  { dappletId: string; userTagId: string },
  { state: RootState; rejectValue: string }
>(
  'auth/removeUserTagFromDapplet',
  async (data, { rejectWithValue, getState }) => {
    const uid = getState().auth.uid
    let stateClone = structuredClone(getState().userData.userDapplets)

    try {
      if (!uid) {
        throw new Error('An error occurred while trying to remove user tag')
      }

      const incomingDappletIndex = stateClone.findIndex(
        dapplet => dapplet.dappletId === data.dappletId,
      )

      if (incomingDappletIndex > -1) {
        stateClone[incomingDappletIndex].userTags = stateClone[
          incomingDappletIndex
        ].userTags.filter(tagId => tagId !== data.userTagId)

        if (
          !stateClone[incomingDappletIndex].dappletState &&
          !stateClone[incomingDappletIndex].userTags.length
        ) {
          stateClone = stateClone.filter(
            dapplet => dapplet.dappletId !== data.dappletId,
          )
        }

        const newData = {
          userDapplets: stateClone,
        }

        await fireStoreSetDoc(newData, 'UsersData', uid, { merge: true })

        return newData
      }
    } catch (e) {
      return rejectWithValue(getErrorMessage(e))
    }
  },
)
