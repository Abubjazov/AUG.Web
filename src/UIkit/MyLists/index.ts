import MyLists from './MyLists'

export default MyLists
