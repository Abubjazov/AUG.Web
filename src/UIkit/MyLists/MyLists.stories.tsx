import type { Meta, StoryObj } from '@storybook/react'
import { mockedReduxProvider as Provider } from 'mockData/mockedReduxProvider'

import MyLists, { MyListsProps } from './MyLists'

import '/src/index.css'

const meta: Meta<MyListsProps> = {
  component: MyLists,
  title: 'UIkit/My Lists',
  parameters: {
    docs: {
      description: {
        component: 'Stub for the MyLists component',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
    menuOpened: { description: 'Hiding the list' },
  },
}

export default meta

type Story = StoryObj<MyListsProps>

export const Default: Story = {
  args: {
    menuOpened: true,
  },
  decorators: [
    Story => (
      <Provider>
        <Story />
      </Provider>
    ),
  ],
}
