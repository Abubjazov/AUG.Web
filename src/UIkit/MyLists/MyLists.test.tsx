import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { fireEvent, render, screen } from '@testing-library/react'
import * as asyncActions from 'asyncThunks/userData'
import {
  mockUserDapplets,
  mockUserLists,
  mockUserTags,
} from 'mockData/mockData'
import { mockedReduxProvider as MProvider } from 'mockData/mockedReduxProvider'
import { Provider } from 'react-redux'
import userDataSliceReducer, {
  EListOperation,
} from 'store/slices/userDataSlice'

import MyLists from './MyLists'

describe('MyLists', () => {
  test('should render MyLists default', () => {
    const { asFragment } = render(
      <MProvider>
        <MyLists menuOpened={true} />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render MyLists when the menu is closed', () => {
    const { asFragment } = render(
      <MProvider>
        <MyLists menuOpened={false} />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render MyLists when the "unInstallModeId"', () => {
    const { asFragment } = render(
      <MProvider>
        <MyLists menuOpened={false} />
      </MProvider>,
    )

    fireEvent.mouseEnter(screen.getByTestId('list-c6rClgOltChco2XnyVDp'))

    expect(asFragment()).toMatchSnapshot()

    fireEvent.mouseLeave(screen.getByTestId('list-c6rClgOltChco2XnyVDp'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render MyLists when the "listOperationGoing"', () => {
    const mockedStore = configureStore({
      reducer: {
        userData: userDataSliceReducer,
      },
      preloadedState: {
        userData: {
          userDapplets: mockUserDapplets,
          userTags: mockUserTags,
          userLists: mockUserLists,
          isAddingUserTag: false,
          isAddingUserList: false,
          isLoadingUserData: false,
          tagOperationGoing: [],
          dappletOperationGoing: [],
          listOperationGoing: [
            {
              listId: 'c6rClgOltChco2XnyVDp',
              operation: EListOperation.REMOVE,
            },
          ],
        },
      },
    })

    const NewProvider = ({ children }: PropsWithChildren<object>) => {
      return <Provider store={mockedStore}>{children}</Provider>
    }

    const { asFragment } = render(
      <NewProvider>
        <MyLists menuOpened={false} />
      </NewProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render MyLists when the "userLists" is empty', () => {
    const mockedStore = configureStore({
      reducer: {
        userData: userDataSliceReducer,
      },
      preloadedState: {
        userData: {
          userDapplets: mockUserDapplets,
          userTags: mockUserTags,
          userLists: [],
          isAddingUserTag: false,
          isAddingUserList: false,
          isLoadingUserData: false,
          tagOperationGoing: [],
          dappletOperationGoing: [],
          listOperationGoing: [],
        },
      },
    })

    const NewProvider = ({ children }: PropsWithChildren<object>) => {
      return <Provider store={mockedStore}>{children}</Provider>
    }

    const { asFragment } = render(
      <NewProvider>
        <MyLists menuOpened={true} />
      </NewProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call installDapplet function on button click, on INSTALL mode', () => {
    const mockedInstallDapplet = vi
      .spyOn(asyncActions, 'removeUserList')
      .mockImplementation(() => vi.fn())

    const { asFragment } = render(
      <MProvider>
        <MyLists menuOpened={true} />
      </MProvider>,
    )

    fireEvent.mouseEnter(screen.getByTestId('list-c6rClgOltChco2XnyVDp'))

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByTestId('delet-list-cross-button'))

    expect(mockedInstallDapplet).toHaveBeenCalledTimes(1)
    expect(mockedInstallDapplet).toHaveBeenCalledWith('c6rClgOltChco2XnyVDp')
  })
})
