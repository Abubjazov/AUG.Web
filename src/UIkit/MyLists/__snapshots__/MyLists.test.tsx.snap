// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyLists > should render MyLists default 1`] = `
<DocumentFragment>
  <div
    class="_root_fa1154  "
  >
    <span
      class="_title_fa1154"
    >
      My Lists
    </span>
    <div
      class="_list_fa1154"
    >
      <div
        class="_list-item_fa1154"
      >
        <span>
          My special list (
          <a
            href="#"
          >
            Me
          </a>
          )
        </span>
      </div>
      <div
        class="_list-item_fa1154"
      >
        <span>
          Financial list (
          <a
            href="#"
          >
            Me
          </a>
          )
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MyLists > should render MyLists when the menu is closed 1`] = `
<DocumentFragment>
  <div
    class="_root_fa1154 _menu-closed_fa1154 "
  >
    <span
      class="_title_fa1154"
    >
      My Lists
    </span>
    <div
      class="_list_fa1154"
    >
      <div
        class="_list-item_fa1154"
      >
        <span>
          My special list (
          <a
            href="#"
          >
            Me
          </a>
          )
        </span>
      </div>
      <div
        class="_list-item_fa1154"
      >
        <span>
          Financial list (
          <a
            href="#"
          >
            Me
          </a>
          )
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
